mne/preprocessing/_optical_density.py

```diff
@@ -0,0 +1,46 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#          Eric Larson <larson.eric.d@gmail.com>
+#
+# License: BSD (3-clause)
+
+from ..io import BaseRaw
+from ..io.constants import FIFF
+from ..utils import _validate_type, warn
+from ..io.pick import _picks_to_idx
+
+import numpy as np
+
+
+def optical_density(raw):
+    r"""Convert NIRS raw data to optical density.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data.
+
+    Returns
+    -------
+    raw : instance of Raw
+        The modified raw instance. Operates in place.
+
+    """
+    _validate_type(raw, BaseRaw, 'raw')
+    picks = _picks_to_idx(raw.info, 'fnirs_raw')
+    data_means = np.mean(raw.get_data(), 1)
+
+    # The devices measure light intensity. Negative light intensities should
+    # not occur. If they do it is likely due to hardware or movement issues.
+    # Set all negative values to abs(x), this also has the benefit of ensuring
+    # that the means are all greater than zero for the division below.
+    if len(np.where(raw._data[picks] <= 0)[0]) > 0:
+        warn("Negative intensities encountered. Setting to abs(x)")
+        raw._data[picks] = np.abs(raw._data[picks])
+
+    for ii in picks:
+        raw._data[ii] /= data_means[ii]
```

Comment:
this will only work if you have preloaded data

Reply:
Thanks, I hadn't considered that. Whats the common approach to this problem? Can I just force the data to be loaded by putting a `raw.load_data()` as the first command? Or is that considered poor form?